tests/iam/test_scps.py

```diff
@@ -0,0 +1,262 @@
+"""
+
+PARENT_ACCOUNT_ID = "745948225562"
+TARGET_ACCOUNT_ID = "593324772711"
+PARENT_ADMIN_ARN = f"arn:aws:iam::{PARENT_ACCOUNT_ID}:role/SCPTestMasterAccount"
+CHILD_ADMIN_ARN = f"arn:aws:iam::{TARGET_ACCOUNT_ID}:role/SCPTestChildSuperRole"
+EXTERNAL_ID = "scps-assume-role-tests"
+# We use hard coded org and ou structure to avoid programmatically creating and tearing down accounts
+# which is not recommended
+OU_ID = "ou-5wjv-glg6vchq"
+FULL_ACCESS_POLICY_ID = "p-FullAWSAccess"
+FULL_ACCESS_POICY_ARN = f"arn:aws:organizations::aws:policy/service_control_policy/{FULL_ACCESS_POLICY_ID}"
+
+
+def assumed_role_client(service_name, target_role_arn, trusted_client):
+    credentials = trusted_client.assume_role(
+        RoleArn=target_role_arn,
+        RoleSessionName="Test",
+        ExternalId=EXTERNAL_ID
+    )["Credentials"]
+
+    return boto3.client(
+        service_name=service_name,
+        aws_access_key_id=credentials['AccessKeyId'],
+        aws_secret_access_key=credentials['SecretAccessKey'],
+        aws_session_token=credentials['SessionToken'],
+    )
+
+def parent_client(service_name):
+    return assumed_role_client(service_name, PARENT_ADMIN_ARN, boto3.client("sts"))
+
+def child_superclient(service_name):
+    return assumed_role_client(service_name, CHILD_ADMIN_ARN, parent_client("sts"))
+
+# Reset the scps for an ou or an account to be just the default allow all
+def reset_scps(TargetId):
+    client = parent_client("organizations")
+    # Start with baseline of only full access
+    try:
+        client.attach_policy(
+            PolicyId=FULL_ACCESS_POLICY_ID,
+            TargetId=TargetId # OU containing child
+        )
+    # It's ok if it's already attached
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "DuplicatePolicyAttachmentException":
+            pass
+        else:
+            raise error
+
+    scps = client.list_policies_for_target(
+        TargetId=TargetId,
+        Filter="SERVICE_CONTROL_POLICY"
+    )["Policies"]
+    # Detach anything other than full access scp
+    for scp in scps:
+        if scp["Arn"] == FULL_ACCESS_POICY_ARN:
+            continue
+        else:
+            client.detach_policy(
+                PolicyId=scp["Id"],
+                TargetId=TargetId
+
+            )
+            client.delete_policy(
+                PolicyId=scp["Id"]
+            )
+    time.sleep(10)
+
+def is_in_parent(ParentId, TargetAccount, client):
+    child_list = client.list_children(
+        ParentId=ParentId,
+        ChildType="ACCOUNT", # Hard coding for now due to the limited use. May pass as an argument if more flexibility is needed
+    )
+    for child in child_list["Children"]:
+        if child["Id"] == TargetAccount:
+            return True
+    # Call again until there are no more results
+    if child_list.get("NextToken"):
+        is_in_parent(ParentId, TargetAccount, client)
```

Comment:
This doesn't appear to ever return anything if it starts paging. Also, there is a paginator available for this: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/organizations.html#Organizations.Paginator.ListChildren